### ISSUE TITLE
[Research] Build Daily Health Summary Pipeline with Mamba-2 Weekly Trend Model

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#26]
## What

Aggregate daily EEG session summaries + Apple Health data into a **structured daily feature vector**, then train a **Mamba-2 weekly model** for longitudinal mental health trend detection. Inspired by Apple's WBM (ICML 2025) and MIT's JETS (NeurIPS 2025 Workshop).

## Why

- Apple's **WBM** (Wearable Behavior Model): trained on 2.5B hours from 161K participants, up to **92% accuracy** on health prediction using daily behavioral summaries
- MIT's **JETS**: trained on 3M person-days from 16K individuals, handles missing data natively, **86.8% AUROC** for health conditions
- Key insight from both papers: **daily-resolution behavioral summaries outperform raw sensor data** for health prediction
- NeuralDreamWorkshop already computes per-session EEG summaries — just need to aggregate daily
- Longitudinal trends (days/weeks) are more predictive of mental health than single-session snapshots

### Step 1: Define daily feature vector (27 features, matching WBM)
```python
# ml/health/daily_summary.py

@dataclass
class DailySummary:
    # EEG session features (averaged across day's sessions)
    avg_valence: float          # Mean valence from emotion classifier
    avg_arousal: float          # Mean arousal
    avg_stress_index: float     # Mean stress
    avg_focus_index: float      # Mean focus
    session_count: int          # Number of EEG sessions today
    total_session_minutes: float
    dominant_emotion: str       # Most frequent emotion
    
    # Apple Health features
    step_count: int
    active_energy_kcal: float
    resting_heart_rate: float
    heart_rate_variability: float  # SDNN or RMSSD
    sleep_duration_hours: float
    sleep_efficiency: float     # time_asleep / time_in_bed
    rem_duration_min: float
    deep_sleep_duration_min: float
    respiratory_rate: float
    blood_oxygen_spo2: float
    
    # Contextual
    day_of_week: int            # 0=Mon, 6=Sun
    is_weekend: bool
    days_since_last_exercise: int
    
    # Questionnaire (when available)
    phq9_score: Optional[int]
    gad7_score: Optional[int]
```

### Step 2: Daily aggregation pipeline
- Cron job (or API endpoint) runs at end of day
- Aggregates all EEG sessions + latest Apple Health data
- Stores in `daily_summaries` database table
- Handles missing data: store None/null, model handles it

### Step 3: Mamba-2 weekly model
```python
# ml/models/weekly_health_model.py
class WeeklyHealthPredictor(nn.Module):
    """Mamba-2 model for weekly health trend prediction.
    
    Input: 7 daily summary vectors (past week)
    Output: next-day predicted mental state + risk scores
    
    Architecture (from Apple WBM):
    - Mamba-2 backbone (linear complexity)
    - Self-supervised pre-training: mask random days, predict them
    - Fine-tune for: PHQ-9 prediction, stress trend, emotion stability
    """
```

### Step 4: Missing data strategy (from JETS/LSM-2)
- Learnable mask tokens for missing modalities
- Train with random modality dropout (30-50%)
- Model learns to predict from whatever data is available
- Days without EEG sessions still have Apple Health data → useful

### Files to Modify
- `ml/health/daily_summary.py` (NEW) — Daily aggregation
- `ml/models/weekly_health_model.py` (NEW) — Mamba-2 model
- `ml/training/train_weekly_model.py` (NEW) — Training script
- `shared/schema.ts` — Add daily_summaries table
- `server/routes.ts` — Add daily aggregation endpoint
- `ml/api/routes.py` — Add `/health/weekly-prediction` endpoint

### Expected Impact
- Longitudinal mental health trend detection (not just single-session)
- "Your stress has been trending up this week" type insights
- Early warning system for depression/anxiety episodes

## Complexity
MEDIUM-HIGH — daily pipeline, Mamba-2 training, needs weeks of data to be useful

## References
- Apple WBM (ICML 2025): https://machinelearning.apple.com/research/beyond-sensor
- JETS (NeurIPS 2025 Workshop): https://openreview.net/forum?id=QqQDjLgHab
- LSM-2 missing data (Google 2025): https://arxiv.org/abs/2506.05321
- Google PHIA (Nature Comms 2025): https://www.nature.com/articles/s41467-025-67922-y
